### PR TITLE
JIP-12 Individual Post Page

### PR DESCRIPTION
**Pull request by dryu99**
Allow users to view individual post pages (by clicking on posts on home page).
Individual pages contain misc post metadata and allow users to view/create/update comments.

Screen Shot 2021-07-21 at 5 25 37 PM

Originally there were issues with naving to individual pages via url (since we do client side routing, the server currently doesn't know how to respond to a client requesting an individual post page). To work around this, we just do a check to see if we naved to the page through client vs url, and fetch data accordingly. We can prob do sth less hacky (like server side rendering oooo) but yknow.

Edit: Ok I didn't notice but since I was testing on non-projected db queries (ack), my code assumed comments existed on all client posts. Which they don't. So now our single post page ALWAYS makes a network request to the server to fetch a post that's populated with comments. niceu

### CODE REVIEW

File: packages/server/src/data/db/db-operations/post-ops.js [hunk 1]
```diff
@@ -61,10 +61,15 @@ const getPostsByFilter = async ({
   return Post.aggregate(aggregationPipeline);
 };
 
+const getPostByID = async (id) => {
+  return await Post.findById(id);
```
Comment (jpfranci) [inline packages/server/src/data/db/db-operations/post-ops.js:65]: Should prob project the same fields as above except with comments

File: packages/client/src/pages/common/nav/nav.tsx [hunk 2]
```diff
@@ -59,6 +63,13 @@ const profileOptions = (
   }
 };
 
+const BrandName = styled.h6`
```
Comment (jpfranci) [inline packages/client/src/pages/common/nav/nav.tsx:66]: Damn h6

**Comment by danyliu9** [comment]
Are we going to have replies on comments?

File: packages/server/src/data/db/db-operations/post-ops.js (hunk 1 above)
Comment (dryu99) [reply packages/server/src/data/db/db-operations/post-ops.js:65]: u right

File: packages/client/src/pages/common/nav/nav.tsx (hunk 2 above)
Comment (dryu99) [reply packages/client/src/pages/common/nav/nav.tsx:66]: hhhhhh

**Comment by dryu99** [comment]
> Are we going to have replies on comments?

Hmm not sure, if we decide to I could def put 'er in, but I'll do it in the next PR I'm gonna work on for comments specifically

**Review by jpfranci — APPROVED** [review]

**Pull request merged**